src/test/scala/org/openkoreantext/processor/tokenizer/KoreanChunkerTest.scala

```diff
@@ -18,10 +18,89 @@
+    val testStr2 = "만약 메일 주소가 하나 있고(예: hong@mail.com) 동시에 수백만명이 메일을 보낸다면 어떻게 될까?"
+    val testStr3 = "트위터 아이디는 언제든지 변경이 가능합니다. @ironman을 @drstrange로 바꿀 수 있는 것이지요."
+    val testStr4 = "구글에는 정말로 이쁜 자전거가 있다. #Google #이쁜자전거 #갖고싶다"
+    val testStr5 = "주식정보 트윗 안내 : Twitter의 주식은 $twtr, Apple의 주식은 $appl 입니다."
+    val testStr6 = "Hey! Can you speak Korean? 한국말! 오케이?"
+    val testStr7 = "ㅋㅋ보다는 ㅎㅎ를 쓰라는데 무슨 차이인가요?"
+    val testStr8 = "6월 21일 개봉한 트랜스포머5:최후의 기사가 혹평 속에서도 박스오피스 1위를 달리고 있다."
+    val testStr9 = "육회가 'six times', 곰탕이 'bear thang' 이라고? 아오! 정말 부끄러운 줄 알아랏!!"
+    val testStr10 = "비가 내리고... 음악이 흐르면... 난 당신을 생각해요~~"
+    
+    assert(
+      findAllPatterns(POS_PATTERNS(URL).matcher(testStr1), URL).mkString("/")
+        === "ChunkMatch(2,32,(http://store.steampowered.com,URL)"
+    )
+    
+    assert(
+      findAllPatterns(POS_PATTERNS(Email).matcher(testStr2), Email).mkString("/")
+        === "ChunkMatch(19,32,hong@mail.com,Email)"
+    )
+    
+    assert(
+      findAllPatterns(POS_PATTERNS(ScreenName).matcher(testStr3), ScreenName).mkString("/")
+        === "ChunkMatch(34,45, @drstrange,ScreenName)/" +
+            "ChunkMatch(24,33, @ironman,ScreenName)"
+    )
+    
+    assert(
+      findAllPatterns(POS_PATTERNS(Hashtag).matcher(testStr4), Hashtag).mkString("/")
+        === "ChunkMatch(35,41, #갖고싶다,Hashtag)/" +
+            "ChunkMatch(28,35, #이쁜자전거,Hashtag)/" +
+            "ChunkMatch(20,28, #Google,Hashtag)"
+    )
+    
+    assert(
+      findAllPatterns(POS_PATTERNS(CashTag).matcher(testStr5), CashTag).mkString("/")
+        === "ChunkMatch(43,49, $appl,CashTag)/" +
+            "ChunkMatch(25,31, $twtr,CashTag)"
+    )
+    
+    assert(
+      findAllPatterns(POS_PATTERNS(Korean).matcher(testStr6), Korean).mkString("/")
+        === "ChunkMatch(32,35,오케이,Korean)/" +
+            "ChunkMatch(27,30,한국말,Korean)"
+    )
+    
+    assert(
+      findAllPatterns(POS_PATTERNS(KoreanParticle).matcher(testStr7), KoreanParticle).mkString("/")
+        === "ChunkMatch(6,8,ㅎㅎ,KoreanParticle)/" +
+            "ChunkMatch(0,2,ㅋㅋ,KoreanParticle)"
+    )
+    
+    assert(
+      findAllPatterns(POS_PATTERNS(Number).matcher(testStr8), Number).mkString("/")
+        === "ChunkMatch(40,41,1,Number)/" +
+            "ChunkMatch(16,17,5,Number)/" +
+            "ChunkMatch(3,6,21일,Number)/" +
+            "ChunkMatch(0,2,6월,Number)"
+    )
+    
+    assert(
+      findAllPatterns(POS_PATTERNS(Alpha).matcher(testStr9), Alpha).mkString("/")
+        === "ChunkMatch(27,32,thang,Alpha)/" +
+            "ChunkMatch(22,26,bear,Alpha)/" +
+            "ChunkMatch(9,14,times,Alpha)/" +
+            "ChunkMatch(5,8,six,Alpha)"
+    )
+    
+    assert(
+      findAllPatterns(POS_PATTERNS(Punctuation).matcher(testStr10), Punctuation).mkString("/")
+        === "ChunkMatch(31,33,~~,Punctuation)/" +
+            "ChunkMatch(17,20,...,Punctuation)/" +
+            "ChunkMatch(6,9,...,Punctuation)"
+    )
```

Comment:
While this is great, you don't have to assign all the strings up front. You can inline all the strings. And the strings do not need to be real strings. For example, in case of `Alpha`, you can just use "민규Hello123" and check if it finds "Hello".